artifactory/system.go

```diff
@@ -33,6 +40,1112 @@ func (v Versions) String() string {
+	oauthProviderSettingsMap := make(map[string]*OauthProviderSetting)
+	if err := unmarshal(oauthProviderSettingsMap); err != nil {
+		return err
+	}
+
+	var oauthProviderSettingsSlice OauthProviderSettings
+	for oauthProviderSettingName, oauthProviderSetting := range oauthProviderSettingsMap {
+		if oauthProviderSetting == nil {
+			continue
+		}
+		oauthProviderSetting.Name = &oauthProviderSettingName
+		oauthProviderSettingsSlice = append(oauthProviderSettingsSlice, oauthProviderSetting)
+	}
+
+	*o = oauthProviderSettingsSlice
+	return nil
+}
+
+// OauthProviderSetting represents the Oauth Provider settings in Artifactory Security Configuration.
+//
+// Docs: https://www.jfrog.com/confluence/display/RTF/YAML+Configuration+File#YAMLConfigurationFile-Security(Generalsecurity,PasswordPolicy,LDAP,SAML,OAuth,HTTPSSO,Crowd)
+type OauthProviderSetting struct {
+	Name         *string `yaml:"-" xml:"name,omitempty"`
+	Id           *string `yaml:"-" xml:"id,omitempty"`
+	ApiUrl       *string `yaml:"apiUrl,omitempty" xml:"apiUrl,omitempty"`
+	AuthUrl      *string `yaml:"authUrl,omitempty" xml:"authUrl,omitempty"`
+	BasicUrl     *string `yaml:"basicUrl,omitempty" xml:"basicUrl,omitempty"`
+	Enabled      *bool   `yaml:"enabled,omitempty" xml:"enabled,omitempty"`
+	ProviderType *string `yaml:"providerType,omitempty" xml:"providerType,omitempty"`
+	Secret       *string `yaml:"secret,omitempty" xml:"secret,omitempty"`
+	TokenUrl     *string `yaml:"tokenUrl,omitempty" xml:"tokenUrl,omitempty"`
+}
+
+// HttpSsoSettings represents the HTTP SSO settings in Artifactory Security Configuration.
+//
+// Docs: https://www.jfrog.com/confluence/display/RTF/YAML+Configuration+File#YAMLConfigurationFile-Security(Generalsecurity,PasswordPolicy,LDAP,SAML,OAuth,HTTPSSO,Crowd)
+type HttpSsoSettings struct {
+	HttpSsoProxied            *bool   `yaml:"httpSsoProxied,omitempty" xml:"httpSsoProxied,omitempty"`
+	RemoteUserRequestVariable *string `yaml:"remoteUserRequestVariable,omitempty" xml:"remoteUserRequestVariable,omitempty"`
+	AllowUserToAccessProfile  *bool   `yaml:"allowUserToAccessProfile,omitempty" xml:"allowUserToAccessProfile,omitempty"`
+	NoAutoUserCreation        *bool   `yaml:"noAutoUserCreation,omitempty" xml:"noAutoUserCreation,omitempty"`
+	SyncLdapGroups            *bool   `yaml:"syncLdapGroups,omitempty" xml:"syncLdapGroups,omitempty"`
+	Reset                     *bool   `yaml:"-" xml:"-"`
+}
+
+// MarshalYAML implements the Marshaller interface.
+func (h HttpSsoSettings) MarshalYAML() (interface{}, error) {
+	if h.Reset != nil && *h.Reset {
+		return nil, nil
+	}
+	return h, nil
+}
+
+// Backup represents the Backup settings in Artifactory Services Configuration.
+//
+// Docs: https://www.jfrog.com/confluence/display/RTF/YAML+Configuration+File#YAMLConfigurationFile-Servicesconfiguration(Backups,MavenIndexer)
+type Backup struct {
+	Key                    *string   `yaml:"-" xml:"key,omitempty"`
```

Comment:
I noticed we're not allowing this field to be serialized in YAML due to the `-` tag.

This peaked my curiosity so was this intentional?

Reply:
Similar to [comment](https://github.com/target/go-arty/pull/57#discussion_r436119180).